air/src/air/trace_info.rs

```diff
@@ -28,6 +28,7 @@ pub struct TraceInfo {
     main_segment_width: usize,
     aux_segment_widths: [usize; NUM_AUX_SEGMENTS],
     aux_segment_rands: [usize; NUM_AUX_SEGMENTS],
+    lagrange_kernel_aux_column_idx: Option<usize>,
```

Comment:
I agree that it doesn't belong in `TraceInfo` - I probably put it there initially to solve some immediate problems.

However, I moved it to `AirContext`, since the Lagrange column index is an Air parameter. Otherwise `AirContext` is incomplete IMO: it holds main/aux transition degrees, the assertions, and trace information, but nothing about the Lagrange column.

Do you agree? I could also add a method `Air::lagrange_kernel_aux_column_idx()` that forwards to the internal context?